index.js

```diff
@@ -231,16 +239,25 @@ const init = async () => {
     });
   });
 
-  // Then we load events, which will include our message and ready event.
-  const evtFiles = await readdir("./events/");
-  evtFiles.forEach(file => {
-    const eventName = file.split(".")[0];
-    client.logger.log(`Loading Event: ${eventName}. 👌`, "log");
-    const event = new (require(`./events/${file}`))(client);
-    // This line is awesome by the way. Just sayin'.
-    client.on(eventName, (...args) => event.run(...args));
-    delete require.cache[require.resolve(`./events/${file}`)];
-  });
+  const eventList = [];
+  klaw("./events").on("data", item => {
+    const { dir, name, ext } = path.parse(item.path);
+    if (!ext || ext !== ".js") return;
+    const eventName = name.split(".")[0];
+    try {
+      const event = new(require(`${dir}${path.sep}${name}${ext}`))(client);
+      eventList.push(event);
+      event.conf.location = dir;
+      client.events.set(event.conf.name, event);
+      client.on(eventName, (...args) => event.run(...args));
+      delete require.cache[require.resolve(`${dir}${path.sep}${name}${ext}`)];
```

Comment:
Not too sure why you would delete the require cache just after setting the events.

Reply:
This is done the same way that it's done with the current guidebot event loader,
```
 // Then we load events, which will include our message and ready event.
  const evtFiles = await readdir("./events/");
  evtFiles.forEach(file => {
    const eventName = file.split(".")[0];
    client.logger.log(`Loading Event: ${eventName}. 👌`, "log");
    const event = new (require(`./events/${file}`))(client);
    // This line is awesome by the way. Just sayin'.
    client.on(eventName, (...args) => event.run(...args));
    delete require.cache[require.resolve(`./events/${file}`)];
  });
  ```
  
  It doesn't affect anything that I've seen so far, but can remove the line if wanted.

Reply:
You raise a very good point, I completely forgot we did that.